engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -890,4 +880,14 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles)(implicit ec: ExecutionContext): Future[WorkflowDescriptor] = {
+    implicit val actorSystem = CromwellServer.actorSystem
+    val materializedWfDescriptorFuture =
+      PromiseActor.askNoTimeout(MaterializeWorkflowDescriptorActor.MaterializeWorkflow(id, workflowSourceFiles), CromwellServer.workflowDescriptorMaterializeActor).mapTo[MaterializationResult]
```

Comment:
Just to make sure I understand correctly, you want a signature like this?

```scala
private def materializeWorkflowDescriptor(
  id: WorkflowId,
  workflowSourceFiles: WorkflowSourceFiles,
  system: ActorSystem,
  actor: ActorRef)(implicit ec: ExecutionContext): Future[WorkflowDescriptor]
```

Then have callers (`getWorkflow`, `getWorkflowByState`) subsequently have the `ActorSystem` and `ActorRef` as parameters too?